src/restruct_poc/beams.hpp

```diff
@@ -0,0 +1,689 @@
+#pragma once
+
+#include <array>
+#include <numeric>
+
+#include <KokkosBlas.hpp>
+
+#include "beams_calcs.hpp"
+#include "interpolation.h"
+#include "types.hpp"
+
+#include "src/gebt_poc/quadrature.h"
+
+namespace oturb {
+
+//------------------------------------------------------------------------------
+// Beam input data structures
+//------------------------------------------------------------------------------
+
+struct BeamNode {
+    double s_;
```

Comment:
Without an accompanying comment describing this attribute, it is not immediately obvious what `s_` represent in a beam node. Please see my next comment on Line 22 for more on this.